specs/version_format_specification.md

```diff
@@ -62,6 +62,13 @@ strings are found to compare as different. In a loop:
    of the whole string), if the other substring has remaining letters, it compares higher.
    Otherwise, the comparison continues at the following characters at point 1.
 
+## Separating the Version from Other String Tokens
+The version is usually part of a larger string, usually containing other tokens such as an
+artifact's name. This specification does not mandate a specific separator between tokens,
+but it suggests to use a character that is not part of the allowed character set that forms
```

Comment:
This sentence is severely misleading. `_` **is** a part of the allowed character set in the version.

I don't think approaching this problem from this side is useful. It's not just a matter of changing a word or two. The basic fact is that there is no unambiguous separator in the common set of characters (`[a-zA-Z0-9._^~+-]`).

Reply:
Yes, this is slopilly worded. When you check the actual algorithm below, it effectively handles `_` as a separator, similarly to `+` and other non-alphanumerical characters. We should update this introductory paragraph.

Reply:
No, they are not ignored. They separate parts of the version string. E.g. `a_b < ab`.

Reply:
> No, they are not ignored. They separate parts of the version string. E.g. `a_b < ab`.

mmh but which part of the spec defines that?

Reply:
This is an iterative algorithm. In step 1, we skip leading characters. We then go to step 2, etc. After iterating though all listed characters classes, we return to step 1. So characters outside of the specified set are all non-sorting separators.

This should be worded better.